prov/efa/test/efa_unit_test_common.c

```diff
@@ -138,7 +138,7 @@ void efa_unit_test_buff_construct(struct efa_unit_test_buff *buff, struct efa_re
 	assert_non_null(buff->buff);
 
 	buff->size = buff_size;
-	err = fi_mr_reg(resource->domain, buff, buff_size, FI_SEND | FI_RECV,
+	err = fi_mr_reg(resource->domain, buff->buff, buff_size, FI_SEND | FI_RECV,
```

Comment:
Interesting. How did it work before?

Reply:
I'm not sure why but when we specify 4096 buff size it always works. When the size is increased to e.g. 8800, it fails ~20% of the time.